specs/ClientCertificate.md

```diff
@@ -0,0 +1,380 @@
+
+                        UINT cert_lst_size;
+                        CHECK_FAILURE(certificateList->get_Count(&cert_lst_size));
+
+                        ICoreWebView2ClientCertificate* certificate = nullptr;
+                        if (cert_lst_size > 0)
+                        {
+                            for (int i = 0; i < cert_lst_size; i++)
+                            {
+                                CHECK_FAILURE(certificateList->GetValueAtIndex(i, &certificate));
+
+                                wil::unique_cotaskmem_string subject;
+                                CHECK_FAILURE(certificate->get_Subject(&subject));
+                                wil::unique_cotaskmem_string issuer;
+                                CHECK_FAILURE(certificate->get_Issuer(&issuer));
+                                wil::unique_cotaskmem_string serialNumber;
+                                CHECK_FAILURE(certificate->get_SerialNumber(&serialNumber));
+                                COREWEBVIEW2_CLIENT_CERTIFICATE_KIND certificateKind;
+                                CHECK_FAILURE(certificate->get_CertificateKind(&certificateKind));
+                                INT64 startDate;
+                                CHECK_FAILURE(certificate->get_ValidStartDate(&startDate));
+                                INT64 expiryDate;
+                                CHECK_FAILURE(certificate->get_ValidExpiryDate(&expiryDate));
+                                ICoreWebView2ClientCertificateIssuerChainList* pemIssuerChainList;
+                                CHECK_FAILURE(certificate->get_PEMEncodedIssuerChainData(&pemIssuerChainList));
+                            }
+                        }
+
+                        if (certificate)
+                        {
+                            CHECK_FAILURE(args->put_SelectedCertificate(certificate));
+                            CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
+                                COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_CERTIFICATE));
+                        }
+                        else
+                        {
+                            CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
+                                COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITHOUT_CERTIFICATE));
+                        }
+                        return S_OK;
+                    })
+                    .Get(),
+                &m_ClientCertificateRequestedToken));
+        }
+        else
+        {
+            CHECK_FAILURE(m_webView->remove_ClientCertificateRequested(
+                m_ClientCertificateRequestedToken));
+        }
+        //! [ClientCertificateRequested]
+    }
+}
+```
+
+## .NET/ WinRT
+```c#
+void ClientCertificateRequestedCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+  try
+  {
+    webView.CoreWebView2.ClientCertificateRequested += delegate (
```

Comment:
Every time we run this command in the sample app we add another ClientCertificateRequested event handler?

Reply:
Okay, added a private member to check whether the event has subscribed to make sure we don't add another event handler if it is